src/Workspaces/Core/Portable/Options/IOptionPersister.cs

```diff
@@ -8,6 +11,7 @@ namespace Microsoft.CodeAnalysis.Options
     /// </summary>
     internal interface IOptionPersister
```

Comment:
I thought of that, and decide not go that way since our option is exposed as Workspace.Options as synchronous API. and that will break all those public API.